src/test/test-lwm2m.c

```diff
@@ -562,94 +1322,522 @@ delete_cb(void *data,
+    }
 
-        objects = sol_lwm2m_client_info_get_objects(cinfo);
+    ASSERT(objects_found == (access_control ? 4 : 3));
+}
 
-        SOL_PTR_VECTOR_FOREACH_IDX (objects, object, i) {
-            uint16_t obj_id;
-            r = sol_lwm2m_client_object_get_id(object, &obj_id);
-            ASSERT(r == 0);
-            if (obj_id == SECURITY_OBJECT_ID ||
-                obj_id == SERVER_OBJECT_ID || obj_id == DUMMY_OBJECT_ID)
-                objects_found++;
-        }
+static void
+nosec_registration_event_cb(void *data, struct sol_lwm2m_server *server,
+    struct sol_lwm2m_client_info *cinfo,
+    enum sol_lwm2m_registration_event event)
+{
+    int r;
+    char *server_type = data;
+
+    if (event == SOL_LWM2M_REGISTRATION_EVENT_REGISTER &&
+        !strcmp("cli1", sol_lwm2m_client_info_get_name(cinfo))) {
+        check_cinfo(cinfo, "cli1", NULL, NULL, SOL_LWM2M_BINDING_MODE_U, true);
+
+        create_obj(server, cinfo, data);
+
+    } else if (event == SOL_LWM2M_REGISTRATION_EVENT_REGISTER) {
+        check_cinfo(cinfo, CLIENT_NAME, SMS_NUMBER, "/my_path", SOL_LWM2M_BINDING_MODE_U, false);
 
-        ASSERT(objects_found == 3);
-        create_obj(server, cinfo);
+        create_obj(server, cinfo, data);
 
     } else if (event == SOL_LWM2M_REGISTRATION_EVENT_UPDATE) {
+        printf("DBG: '%s' ---[Unobserve /999/0/2]---> '%s'\n",
+            server_type, sol_lwm2m_client_info_get_name(cinfo));
         r = sol_lwm2m_server_del_observer(server, cinfo, "/999/0/2",
-            observe_res_cb, NULL);
+            observe_res_cb, data);
         ASSERT(r == 0);
+
+        printf("DBG: '%s' ---[Delete /999/0]---> %s\n",
+            server_type, sol_lwm2m_client_info_get_name(cinfo));
         r = sol_lwm2m_server_delete_object_instance(server, cinfo, "/999/0",
-            delete_cb, NULL);
+            delete_cb, data);
+        ASSERT(r == 0);
+    } else {
+        ASSERT(1 == 2); //TIMEOUT/UNREGISTER, this must not happen!
+    }
+}
+
+#ifdef DTLS
+static void
+write_acl_unauthorized_cb(void *data,
+    struct sol_lwm2m_server *server,
+    struct sol_lwm2m_client_info *client, const char *path,
+    enum sol_coap_response_code response_code)
+{
+    int r;
+    char *server_type = data;
+
+    ASSERT(response_code == SOL_COAP_RESPONSE_CODE_UNAUTHORIZED);
+
+    printf("DBG: '%s' ---[Delete /]---> %s\n",
+        server_type, sol_lwm2m_client_info_get_name(client));
+    r = sol_lwm2m_server_delete_object_instance(server, client, "/",
+        delete_cb, data);
+    ASSERT(r == -EINVAL);
+
+    delete_cb(data, server, client, NULL, SOL_COAP_RESPONSE_CODE_UNAUTHORIZED);
+}
+
+static void
+read_unauthorized_cb(void *data,
+    struct sol_lwm2m_server *server,
+    struct sol_lwm2m_client_info *client,
+    const char *path,
+    enum sol_coap_response_code response_code,
+    enum sol_lwm2m_content_type content_type,
+    struct sol_str_slice content)
+{
+    ASSERT(response_code == SOL_COAP_RESPONSE_CODE_UNAUTHORIZED);
```

Comment:
Why? I mean, I'm only doing this "hang around" thing for the negative- and corner-cases. Since they should fail anyway there's no dependency between the operations (like a Create needing to be executed before a Read can happen and so on).